libblkid/src/partitions/mac.c

```diff
@@ -123,12 +123,12 @@ static int probe_mac_pt(blkid_probe pr,
 	ssf = block_size / 512;
 	nblks = be32_to_cpu(p->map_count);
 
```

Comment:
Linux kernel is more robust in this case, it uses:
```
        blocks_in_map = be32_to_cpu(part->map_count);
        if (blocks_in_map < 0 || blocks_in_map >= DISK_MAX_PARTS) {
                put_dev_sector(sect);
                return 0;
        }
      
        if (blocks_in_map >= state->limit)
                blocks_in_map = state->limit - 1;
```
where DISK_MAX_PARTS is 256 and state->limit is the current limit for the device. I think use add `if (nblks > 256) goto nothing;` would be a good solution for libblkid too.

Reply:
Thank you for your great idea to look up in kernel source! I have adjusted atari.c and mac.c to consider limits.